src/XIOPAK.CPP

```diff
@@ -176,64 +176,32 @@ RC FC xfGetPath(	// Check path / retrieve full pathname of a file
 
 // Returns RCOK if ok, with *fnamp replaced, else RCBAD.
 {
-	const char *fname, *p, *namepart, *dir;
-	SEC sec;
-	char save, cwdSave[ _MAX_PATH];			// save current dir pathname
-
+	filesys::path fname(*fnamp), dir;	// filesystem variables
+	std::error_code ec;					// Error Code for filesystem functions
 
-	fname = strupr( strTrim( NULL, *fnamp) );		// upper case and deblank
-	// uppercasing appears UNNECESSARY. ? xx
-
-	p = strrchr(fname,'\\');		// find last \, NULL if no directories
-	if ( p != NULL				// if contains '\'
-			|| (p=strrchr(fname,':')) != NULL ) 	// or else colon
-	{
-		namepart = p+1;		// name.ext starts after drv:dir\
-
-		// substring for selecting drv:dir includes colon and leading (root) '\'
-		if (*p==':'		// if drive only, no \'s
-				|| *(p-1)==':'  	// or just root after drive (only 1 \)
-				|| (p-fname)==0)	// or root only (or : - bad syntax)
-			p++;			// include : or leading \ in directory string
-		// else: no p++: \ after (last) dir name is omitted.
-
-		/* select drive/dir: easy way to interpret ..\'s, etc, to get full path.
-				also verifies existence of drive and directories. */
-		save = *p;
-		*(char *)p = '\0';		// terminate before name.ext for chdir. override const 10-94 -- but we restore it.
-		getcwd( cwdSave, _MAX_PATH );	// save drv:dir in stack array
-		sec = xchdir( fname, IGN);	// select path's drv:dir. suppress message.
-		if (sec != SECOK)		// if chdir failed
-			// xchdir's "cannot change working directory to %s" message is inappropriate eg when checking CSE input path.
-			switch (sec)			// issue our own more appropariate message. err() is in rmkerr.cpp.
-			{
-			case SECBADRV:
-				err( erOp, "\"%s\": Invalid drive", fname);
-				break;	// (d:dirpath displays -- accept)
-			case ENOENT:
-				err( erOp, "\"%s\": Directory not found", fname);
-				break;
-			default:
-				err( erOp, "\"%s\": Invalid path (sec=%d)", fname, INT(sec));
-				break;	// not expected
-			}
-		dir = xgetdir(); 		// get full drv:dir
-
-		xchdir( cwdSave, ABT );		// restore drv:dir; err impossible?
-		*(char *)p = save;		// replace char
-		if (sec != SECOK)		// now, if chdir failed
-			return RCBAD;			// give bad return
+	if (filesys::exists(fname, ec)) {	// File or full path exist
+		dir = filesys::absolute(fname); // Create full path
 	}
-	else	// no drive, no directory
-	{
-		namepart = fname;
-		dir = xgetdir();		// get full current drive/dir
+	else {
+		if (!fname.has_root_name()) {			// Missing root name, ex C:
```

Comment:
I think we should understand the values of `ec` that are returned and use it directly instead. We maybe don't even need each of these separate checks because the `std::error_code` should contain information that we can pass back to the user. This is probably the case for many of our uses of `std::error_code`.

Reply:
@galanCA did you look at other uses of `std::error_code` to see if we want to change the messages there too?